### DISCUSSION TITLE
closeModalWithEvents() not working

### DISCUSSION THREAD

**Discussion by DonCamillo11** [wire-elements/modal#422] [Q&A]
Hi,
I have a modal from which I open a child modal. When saving, I want to close the modal using closeModalWithEvents() and update a select field in the parent modal. This always worked without any problems in others, but I can't get it to work with this one. In the child component I have
```
$this->closeModalWithEvents([
    JobTimeCreate::class => 'updateTaskList',
]);
```
and in the parent component
```
#[On('updateTaskList')]
public function updateTaskList(){
    $this->jobtasks = JobTask::where('job_id', $this->job_id)->orderBy('title')->get();
}
```
I figured out that this function never gets called. I don’t know why?

**Comment by tonyuh** [comment]
Hi,

I'm having the same issue, how do I close the modal so that I can pass data back to the parent component? Seems like the listener on the parent component doesn't get triggered. Is there a working example of this function? I think there might be extra 'use' that I am missing.

**Comment by NoChizPlz** [comment]
any update on this? i cant refresh my parent modal

**Reply by tonyuh** [reply]
I ended up doing 2 lines. First to dispatch then the second to close the modal.

so this
```
$this->closeModalWithEvents([
    JobTimeCreate::class => 'updateTaskList',
]);
```
becomes
```
$this->dispatch('updateTaskList');
$this->closeModal();
```
you can pass variable back by adding an argument like this
```
$this->dispatch('updateTaskList', parent_variable_name: $child_variable_name);
$this->closeModal();
```

**Reply by DonCamillo11** [reply]
I used the same code too
```
$this->dispatch('updateTaskList');
$this->closeModal();
```
